### PR TITLE
Fix asfarray kwarg default handling.

### LINKED ISSUE #6449: `asfarray` incorrectly handles the default kwarg
- [x] I have tried using the 0.52.0rc2 released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

This:
```python
from numba import njit
import numpy as np

@njit
def foo():
    np.asfarray(np.zeros(4,))

foo()
```
produces:
```
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<function asfarray at 0x7ff2ebd55ef0>) found for signature:
 
 >>> asfarray(array(float64, 1d, C))
 
There are 2 candidate implementations:
  - Of which 2 did not match due to:
  Overload in function 'np_asfarray': File: numba/np/arraymath.py: Line 4089.
    With argument(s): '(array(float64, 1d, C))':
   Rejected as the implementation raised a specific error:
     NotImplementedError: <class 'numpy.float64'> cannot be represented as a Numpy dtype
       Traceback (most recent call last):
         File "<path>/numba/core/typing/templates.py", line 677, in _get_impl
           impl, args = self._impl_cache[cache_key]
       KeyError: (<numba.core.typing.context.Context object at 0x7ff2dccf0050>, (array(float64, 1d, C),), ())
       During handling of the above exception, another exception occurred:
       Traceback (most recent call last):
         File "<path>/numba/core/types/functions.py", line 298, in get_call_type
           sig = temp.apply(nolitargs, nolitkws)
         File "<path>/numba/core/typing/templates.py", line 345, in apply
           sig = generic(args, kws)
         File "<path>/numba/core/typing/templates.py", line 597, in generic
           disp, new_args = self._get_impl(args, kws)
         File "<path>/numba/core/typing/templates.py", line 679, in _get_impl
           impl, args = self._build_impl(cache_key, args, kws)
         File "<path>/numba/core/typing/templates.py", line 710, in _build_impl
           ovf_result = self._overload_func(*args, **kws)
         File "<path>/numba/np/arraymath.py", line 4091, in np_asfarray
           dtype = as_dtype(dtype)
         File "<path>/numba/np/numpy_support.py", line 152, in as_dtype
           % (nbtype,))
       NotImplementedError: <class 'numpy.float64'> cannot be represented as a Numpy dtype
  raised from <path>/numba/np/numpy_support.py:152

During: resolving callee type: Function(<function asfarray at 0x7ff2ebd55ef0>)
During: typing of call at test_farray.py (6)


File "test_farray.py", line 6:
def foo():
    np.asfarray(np.zeros(4,))
    ^
```

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Fixes #6449

### CODE REVIEW

File: numba/np/arraymath.py [hunk 1]
```diff
@@ -4088,7 +4088,10 @@ def impl(a, dtype=None):
 
 @overload(np.asfarray)
 def np_asfarray(a, dtype=np.float64):
-    dtype = as_dtype(dtype)
+
```
Comment (sklam) [inline numba/np/arraymath.py:4091]: nitpick: extra line

**Review by sklam — COMMENTED** [review]
Thanks for the patch. I've just one very minor nitpick.

File: numba/np/arraymath.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/np/arraymath.py]: fixed in 0e43734

**Review by sklam — APPROVED** [review]

**Pull request merged**